transaction.go

```diff
@@ -1,16 +1,39 @@
 package makroud
 
 import (
+	"context"
+	"database/sql"
+
 	"github.com/pkg/errors"
 )
 
+// TxOptions is an alias for sql.TxOptions to reduce import leak.
+// This alias allows the use of makroud.TxOptions and sql.TxOptions seamlessly.
+type TxOptions = sql.TxOptions
+
+// List of supported isolation level for a postgres transaction.
+const (
+	LevelDefault         = sql.LevelDefault
+	LevelReadUncommitted = sql.LevelReadUncommitted
+	LevelReadCommitted   = sql.LevelReadCommitted
+	LevelRepeatableRead  = sql.LevelRepeatableRead
+	LevelSerializable    = sql.LevelSerializable
+)
+
 // Transaction will creates a transaction.
 func Transaction(driver Driver, handler func(driver Driver) error) error {
```

Comment:
@thoas But keep this one as is.

Reply:
Or maybe add a context, such as `makroud.Exec(ctx, driver, ...)`

Reply:
if we do want to provide a context directly in `BeginContext`, we can break the signature but `opts *TxOptions` should be an optional parameter, I'm not a big fan of duplicating `Transaction` as `TransactionWithOptions` maybe we can use variadic here?